projects/zopfli/zopfli_compress_fuzzer.cc

```diff
@@ -0,0 +1,30 @@
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <string>
+
+#include "zopfli.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  ZopfliOptions options;
+  ZopfliInitOptions(&options);
+
+  if (size == 0 || size > 1024) {
+    return 0;
+  }
+
+  const ZopfliFormat format = static_cast<ZopfliFormat>(data[0] % 3);
+  data++;
+  size--;
+
+  unsigned char* outbuf = nullptr;
+  size_t outsize = 0;
+
+  ZopfliCompress(&options, format, data, size, &outbuf, &outsize);
```

Comment:
This implementation still has a problem: if `ZopfliCompress` has a buffer underflow bug and attempts to access the `data - 1` address, the sanitizers will not report a problem, as that address us actually a part of the same allocation. So, instead of doing `data++` on line 17, we'd better copy the rest of the data into a new buffer. To avoid extra complications here, I'd recommend bringing FuzzedDataProvider back and use it as you had previously: pick a value from an array and then `ConsumeRemainingBytes` as an input.

Reply:
Yes, I agree. The change has been made. I changed initially because the developer of zopfli didn't want to include an extra library. However, the speed does not seem to be any better.

Reply:
Thanks a lot!